test/integration_test.py

```diff
@@ -206,19 +206,20 @@ def _public_tdr_client(self) -> TDRClient:
 
     @cached_property
     def _unregistered_tdr_client(self) -> TDRClient:
-        tdr = TDRClient(
-            credentials_provider=ServiceAccountCredentialsProvider(
-                service_account=config.ServiceAccount.unregistered
-            )
+        credentials_provider = ServiceAccountCredentialsProvider(
+            service_account=config.ServiceAccount.unregistered
         )
+        tdr = TDRClient(credentials_provider=credentials_provider)
         email = tdr.credentials.service_account_email
         self.assertFalse(tdr.is_registered(),
                          f'The "unregistered" service account ({email!r}) has '
                          f'been registered')
-        # The unregistered service account should not have access to any sources
-        # FIXME: IT assertion for snapshot listing with unreg SA conflates 401 and 500
-        #        https://github.com/DataBiosphere/azul/issues/4086
-        self.assertRaises(RequirementError, tdr.snapshot_names_by_id)
+        expected_access_error = credentials_provider.insufficient_access(
```

Comment:
This looks like you're deriving the expectation using the code under test which is a smell. If there were bugs in `insufficient_access` we risk a false negative.